src/payment-request.ts

```diff
@@ -0,0 +1,77 @@
+import { gql, GraphQLClient } from "graphql-request";
+import { getSessionById, checkSessionLock } from "./session";
+import keyPathAccessor from "lodash.property";
+import setByKeyPath from "lodash.set";
+import type { PaymentRequest, Session, KeyPath, Value } from "./types";
+
+export async function createPaymentRequest(
+  client: GraphQLClient,
+  {
+    sessionId,
+    sessionPreviewKeys,
+    payeeName,
+    payeeEmail,
+  }: {
+    sessionId: string;
+    sessionPreviewKeys: KeyPath[];
+    payeeName: string;
+    payeeEmail: string;
+  }
+): Promise<PaymentRequest> {
+  const isSessionLocked = await checkSessionLock(client, sessionId);
+  if (!isSessionLocked) {
+    throw new Error(
+      "session mush be locked before a payment request can be created"
+    );
+  }
+  const session = await getSessionById(client, sessionId);
+  const sessionPreviewData: PaymentRequest["sessionPreviewData"] =
+    buildSessionPreviewData(session, sessionPreviewKeys); // throws if sessionPreviewData cannot be built
```

Comment:
Building the session preview data from keys ensures that the data exists on the session. The client can specify the keys but the API doesn't need to trust client data directly.